com.unity.cinemachine/Runtime/Helpers/InputAxisController.cs

```diff
@@ -274,6 +281,39 @@ static float ReadLegacyInput(string axisName)
         internal static SetControlDefaultsForAxis SetControlDefaults;
 
 #if CINEMACHINE_UNITY_INPUTSYSTEM
+        /// <summary>
+        /// Definition of how we read input. Override this in your child classes to specify
+        /// the InputAction's type to read if it is different from Vector2.
+        /// </summary>
+        /// <param name="action">The action being read.</param>
+        /// <param name="hint">The axis hint of the action.</param>
+        /// <returns>Returns the value of the input device.</returns>
+        protected virtual float ReadInput(InputAction action, IInputAxisSource.AxisDescriptor.Hints hint)
+        {
+            var activeControl = action.activeControl;
+            if (activeControl == null)
+                return 0f;
+            
+            var actionControlType = activeControl.valueType;
+            if (actionControlType == typeof(float))
+                return action.ReadValue<float>();
+            if (actionControlType == typeof(Vector2))
+                switch (hint)
+                {
+                    default:
+                    case IInputAxisSource.AxisDescriptor.Hints.Default:
```

Comment:
This looks bigger than it needs to be.  Why not just this:
if (hint is Y) return Y component;
return X component